nemo/collections/common/tokenizers/huggingface/auto_tokenizer.py

```diff
@@ -0,0 +1,213 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+from typing import Dict, List, Optional
+
+from transformers import ALL_PRETRAINED_CONFIG_ARCHIVE_MAP
+from transformers import AutoTokenizer as AUTOTOKENIZER
+
+from nemo.collections.common.tokenizers.tokenizer_spec import TokenizerSpec
+from nemo.utils import logging
+
+__all__ = [
+    'AutoTokenizer',
+]
+
+
+def handle_quotes(text):
+    text_ = ""
+    quote = 0
+    i = 0
+    while i < len(text):
+        if text[i] == "\"":
+            if quote % 2:
+                text_ = text_[:-1] + "\""
+            else:
+                text_ += "\""
+                i += 1
+            quote += 1
+        else:
+            text_ += text[i]
+        i += 1
+    return text_
+
+
+def remove_spaces(text):
+    text = text.replace("( ", "(")
+    text = text.replace(" )", ")")
+    text = text.replace("[ ", "[")
+    text = text.replace(" ]", "]")
+    text = text.replace(" / ", "/")
+    text = text.replace("„ ", "„")
+    text = text.replace(" - ", "-")
+    text = text.replace(" ' ", "'")
+    text = re.sub(r'([0-9])( )([\.,])', '\\1\\3', text)
+    text = re.sub(r'([\.,])( )([0-9])', '\\1\\3', text)
+    text = re.sub(r'([0-9])(:)( )([0-9])', '\\1\\2\\4', text)
+    text = text.replace(" %", "%")
+    text = text.replace("$ ", "$")
+    text = re.sub(r'([^0-9])(,)([0-9])', '\\1\\2 \\3', text)
+    return text
+
+
+class AutoTokenizer(TokenizerSpec):
+    '''
+        Wrapper of HuggingFace AutoTokenizer https://huggingface.co/transformers/model_doc/auto.html#autotokenizer.
+    '''
+
+    def __init__(
+        self,
+        pretrained_model_name: str,
```

Comment:
or we can set a WPE default here @yzhang123 , @titu1994 ?

Reply:
we can set "bert-base-cased" as default

Reply:
if @titu1994 doesn't need a default, I'm fine with not having it

Reply:
ill leave it as required. Our configs set the default, i think that is a good trade off, so its visible to user